bundle/config/mutator/python/python_locations.go

```diff
@@ -0,0 +1,163 @@
+package python
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/databricks/cli/libs/dyn"
+)
+
+const generatedFileName = "__generated_by_pydabs__.yml"
+
+// pythonLocations is data structure for efficient location lookup for a given path
```

Comment:
This can be expanded. The PR summary explains the need for this trie well. Maybe copy it here?